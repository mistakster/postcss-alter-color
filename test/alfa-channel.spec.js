const postcss = require('postcss');
const alterColorPlugin = require('../lib/index');

describe('Alter Color plugin', () => {

  test('should alter simple color alpha channel in a single property', () => {
    return postcss()
      .use(alterColorPlugin({from: 'black', to: 'red', preserveAlphaChannel: true}))
      .process(`div{color:rgba(0,0,0,0.5)}`, {from: undefined})
      .then(result => {
        expect(result.css).toBe(`div{color:rgba(255,0,0,0.5)}`)
      });
  });

  test('should alter simple color without alpha channel in a single property', () => {
    return postcss()
      .use(alterColorPlugin({from: 'black', to: 'red'}))
      .process(`div{color:rgba(0,0,0,0.5)}`, {from: undefined})
      .then(result => {
        expect(result.css).toBe(`div{color:rgba(0,0,0,0.5)}`)
      });
  });

  test('should alter simple color alpha channel in a single property', () => {
    return postcss()
      .use(alterColorPlugin({from: 'black', to: 'red', preserveAlphaChannel: true}))
      .process(`div{color:rgba(0,0,0,0.5);outline:1px solid hsla(0,0%,0%,0.9);border:solid 1px rgb(0,0,0)}`, {from: undefined})
      .then(result => {
        expect(result.css).toBe(`div{color:rgba(255,0,0,0.5);outline:1px solid hsla(0,100%,50%,0.9);border:solid 1px rgb(255,0,0)}`)
      });
  });
});
