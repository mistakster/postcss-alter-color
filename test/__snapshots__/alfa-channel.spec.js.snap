// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should alter a color and leave alpha channel untouched 1`] = `
"
div {
  color: rgba(255,0,0,0.5);
}
"
`;

exports[`should alter both rgb(), rgba(), hsl() and hsla() colors 1`] = `
"
div {
  border: solid 1px rgb(255,0,0);
  color: rgba(255,0,0,0.5);
  background-color: hsl(0,100%,50%);
  outline: 1px solid hsla(0,100%,50%,0.9);
}
"
`;

exports[`should not alter a color with different alpha channel 1`] = `
"
div {
  color: rgba(0,0,0,0.5);
}
"
`;
